src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ClientConnectionManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Web;
+using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Hosting;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.Owin;
+
+namespace Microsoft.Azure.SignalR.AspNet
+{
+    internal class ClientConnectionManager : IClientConnectionManager
+    {
+        private static readonly ClaimsPrincipal EmptyPrincipal = new ClaimsPrincipal(new ClaimsIdentity());
+
+        private static readonly string[] SystemClaims =
+        {
+            "aud", // Audience claim, used by service to make sure token is matched with target resource.
+            "exp", // Expiration time claims. A token is valid only before its expiration time.
+            "iat", // Issued At claim. Added by default. It is not validated by service.
+            "nbf"  // Not Before claim. Added by default. It is not validated by service.
+        };
+
+        private readonly PersistentConnection _dispatcher;
+
+        public ClientConnectionManager(IDependencyResolver resolver)
+        {
+            _dispatcher = resolver.Resolve<PersistentConnection>() ?? throw new ArgumentNullException(nameof(resolver));
```

Comment:
This doesn't look right. Why are we resolving a PersistentConnection here?

Reply:
used to `_dispatcher.Authorize` and `_dispatcher.ProcessRequest` when `CreateConnection` https://github.com/Azure/azure-signalr/pull/159/files/74cd48519753460f87f4e030b18068447ef66576#diff-4a1af73f4f7a69cde0fb2ba90b135438R44

Reply:
I think we need a dispatcher but this shouldn't be in the IDependencyResolver. It's the wrong type.

Reply:
We should be creating one of these per connection, not resolving it from the container.

Reply:
Sure